expo/expo
# feat(cli, router, babel): add router root to metro cache

# Why

- Add support for invalidating the cache when the app directory changes. We use this feature in our E2E tests so it would be good to ensure it works properly.
- This change also enables us to move the Expo config read operation out of the babel preset which should generally be safer and faster.
- By adding this invalidation we also ensure that the cache is cleared when expo-router is installed, this should help reduce issues related to manual expo-router installation in existing projects.
- This change also deletes `expo-router/babel` in favor of `babel-preset-expo` which is properly tested.
- I've also added the option to change the root directory to the docs and noted that it's a dangerous operation that may lead to issue closing.

# Test Plan

- Ensure project continue working.
- Ensure the cache clears when the root directory changes.



# Checklist
